source/common/network/filter_state_proxy_info.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "envoy/network/address.h"
+#include "envoy/stream_info/filter_state.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * Information which filters can add if they detect the stream should go
+ * upstream through an HTTP/1.1 proxy.
+ */
+class FilterStateProxyInfo : public StreamInfo::FilterState::Object {
```

Comment:
nit: It looks like the naming convention for such object is usually <Something>FilterState. So how about "ProxyInfoFilterState"?